Plugins/NativeCamera/Source/MediaStream.cpp

```diff
@@ -58,13 +58,21 @@ namespace Babylon::Plugins
 
     MediaStream::~MediaStream()
     {
+        m_cancellationSource.cancel();
+
+        // HACK: This is a hack to make sure the camera device is destroyed on the JS thread.
+        // The napi-jsi adapter currently calls the destructors of JS objects possibly on the wrong thread.
+        // Once this is fixed, this hack will no longer be needed.
```

Comment:
Do we have/need an issue created to address this hack in the future?

Reply:
I didn't find one. I will create one.